moose/src/textual/parsing.rs

```diff
@@ -172,9 +172,9 @@ fn parse_assignment<'a, E: 'a + ParseError<&'a str> + ContextError<&'a str>>(
 ) -> IResult<&'a str, Operation, E> {
     let (input, identifier) = ws(identifier)(input)?;
     let (input, _) = tag("=")(input)?;
-    let (input, operator) = ws(parse_operator)(input)?;
+    let (input, operator) = ws(cut(context("Expecting operator name", parse_operator)))(input)?;
```

Comment:
The cut-contex makes the parser to avoid trying out any other parsers that may be in the scope (via `alt`, for example). It bubbles up the error.

In this case, the parser already seen `x = ` part of the string, so it looks like a correct assignment. But then the operator parsing failed - there is no reason to try something else and is safe to tell user the error.

The context provides more information to the error that bubbles up.

Before this change, most errors in the input file lead to a misleading `expecting EOF` error. That was because one of the parsers in scope is to allow empty (whitespaces only) lines in the end of the input and it was its error that ended up being reported in many cases. The cut-context makes the better error to pop up instead.